python/mxnet/kvstore.py

```diff
@@ -320,18 +320,20 @@ def num_workers(self):
         check_call(_LIB.MXKVStoreGetGroupSize(self.handle, ctypes.byref(size)))
         return size.value
 
-    def save_optimizer_states(self, fname):
+    def save_optimizer_states(self, fname, with_optimizer=False):
```

Comment:
dump_optimizer.
explain in comment what's the difference between states and optimizer.

like:
Saves the optimizer (updater) state (momentum etc)

dump_optimizer:
whether to also serialize the optimizer itself. This would also save learning rate schedule etc